examples/ASDKgram/Sample/CommentFeedModel.m

```diff
@@ -0,0 +1,196 @@
+{
+  return [_comments objectAtIndex:index];
+}
+
+- (NSUInteger)numberOfCommentsForPhoto
+{
+  return _totalItems;
+}
+
+- (BOOL)numberOfCommentsForPhotoExceedsInteger:(NSUInteger)number
+{
+  return (_totalItems > number);
+}
+
+- (NSAttributedString *)viewAllCommentsAttributedString
+{
+  NSString *string               = [NSString stringWithFormat:@"View all %@ comments", [NSNumber numberWithUnsignedInteger:_totalItems]];
+  NSAttributedString *attrString = [NSAttributedString attributedStringWithString:string fontSize:14 color:[UIColor lightGrayColor] firstWordColor:nil];
+  return attrString;
+}
+
+- (void)requestPageWithCompletionBlock:(void (^)(NSArray *))block
+{
+  // only one fetch at a time
+  if (_fetchPageInProgress) {
+      return;
+  } else {
+    _fetchPageInProgress = YES;
+    [self fetchPageWithCompletionBlock:block];
+  }
+}
+
+- (void)refreshFeedWithCompletionBlock:(void (^)(NSArray *))block
+{
+  // only one fetch at a time
+  if (_refreshFeedInProgress) {
+    return;
+  } else {
+    _refreshFeedInProgress = YES;
+    _currentPage = 0;
+    
+    // FIXME: blow away any other requests in progress
+    
+    [self fetchPageWithCompletionBlock:^(NSArray *newPhotos) {
+      if (block) {
+        block(newPhotos);
+      }
+      _refreshFeedInProgress = NO;
+    } replaceData:YES];
+  }
+}
+
+#pragma mark - Helper Methods
+- (void)fetchPageWithCompletionBlock:(void (^)(NSArray *))block
+{
+  [self fetchPageWithCompletionBlock:block replaceData:NO];
+}
+
+- (void)fetchPageWithCompletionBlock:(void (^)(NSArray *))block replaceData:(BOOL)replaceData
+{
+  // early return if reached end of pages
+  if (_totalPages) {
+    if (_currentPage == _totalPages) {
+      return;
+    }
+  }
+  
+  dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+    
+    NSMutableArray *newComments = [NSMutableArray array];
+    
+    @synchronized(self) {
+      
+      NSUInteger nextPage = _currentPage + 1;
+      
+      NSString *urlAdditions = [NSString stringWithFormat:@"page=%lu", (unsigned long)nextPage];
+      NSURL *url = [NSURL URLWithString:[_urlString stringByAppendingString:urlAdditions]];
+      
+      NSData *data = [NSData dataWithContentsOfURL:url];
+      
+      if (data) {
+        
+        NSDictionary *response = [NSJSONSerialization JSONObjectWithData:data options:0 error:NULL];
+        
+        if ([response isKindOfClass:[NSDictionary class]]) {
+          
+          _currentPage = [[response valueForKeyPath:@"current_page"] integerValue];
+          _totalPages  = [[response valueForKeyPath:@"total_pages"] integerValue];
+          _totalItems  = [[response valueForKeyPath:@"total_items"] integerValue];
+          
+          NSArray *comments = [response valueForKeyPath:@"comments"];
+          
+          if ([comments isKindOfClass:[NSArray class]]) {
+            
+            NSUInteger numComments = [comments count];
+            if (numComments > 3) {
```

Comment:
Haha. I am in the same boat when it comes to being new to both iOS and ASDK. :) 

I have been ask to present ASDK to the senior IOS devs in two weeks, so your sample came at the right time. The first impression from running it on my 6s+ was that was that UIKit version looked perceivably the same. FPS got to the 30s once in a while, but not too noticeable to the eye. So of course I had to see what I could do to make the app more real world with UIKit being slower than ASDK. ;)